src/shared/libfido2-util.c

```diff
@@ -520,6 +636,32 @@ int fido2_use_hmac_hash(
                         goto finish;
                 }
 
+                bool in_token = false;
+
+                r = fido2_is_cred_in_specific_token(path, rp_id, cid, cid_size, pins, required, &in_token);
+                /* We handle PIN related errors here since we have used PIN in the check.
+                 * If PIN error happens, we can avoid pin retries decreased the second time. */
+                if (r < 0) {
```

Comment:
we like shallow indentation, hence just drop this `if (r < 0)` early check here, just check against the various values in the if checks below.